client/src/modules/Course/Components/Gauges.tsx

```diff
@@ -0,0 +1,189 @@
+        if (overall) {
+            if (0 <= overall && overall < 3) {
+                if (overall < 0.25) {
+                    setStars([Gray, Gray, Gray, Gray, Gray])
+                } else if (0.25 <= overall && overall < 0.8) {
+                    setStars([RedHalf, Gray, Gray, Gray, Gray])
+                } else if (0.8 <= overall && overall < 1.25) {
+                    setStars([Red, Gray, Gray, Gray, Gray])
+                } else if (1.25 <= overall && overall < 1.8) {
+                    setStars([Red, RedHalf, Gray, Gray, Gray])
+                } else if (1.8 <= overall && overall < 2.25) {
+                    setStars([Red, Red, Gray, Gray, Gray])
+                } else {
+                    setStars([Red, Red, RedHalf, Gray, Gray])
+                }
+            } else if (3 <= overall && overall < 3.8) {
+                if (overall < 3.25) {
+                    setStars([Yellow, Yellow, Yellow, Gray, Gray])
+                } else {
+                    setStars([Yellow, Yellow, Yellow, YellowHalf, Gray])
+                }
+            } else {
+                if (overall < 4.25) {
+                    setStars([Green, Green, Green, Green, Gray])
+                } else if (overall < 4.8) {
+                    setStars([Green, Green, Green, Green, GreenHalf])
+                } else {
+                    setStars([Green, Green, Green, Green, Green])
+                }
+            }
+        } else {
+            setStars([Gray, Gray, Gray, Gray, Gray])
+        }
+    }, [overall])
+
+    useEffect(() => {
+        // Higher difficulty scores are more negative
+        if (difficulty) {
+            if (0 <= difficulty && difficulty < 3) {
+                setDifficultyEmote(HappyFace)
+                setDifficultyHover('Super easy')
+                setDifficultyColor(green)
+            } else if (3 <= difficulty && difficulty < 3.8) {
+                setDifficultyEmote(MehFace)
+                setDifficultyHover('Doable')
+                setDifficultyColor(yellow)
+            } else {
+                setDifficultyEmote(SadFace)
+                setDifficultyHover('Challenging')
+                setDifficultyColor(red)
+            }
+            setDifficultyBars([
+                (Math.min(difficulty, 1) * 100).toFixed(0) + '%',
+                (Math.min(difficulty - 1, 1) * 100).toFixed(0) + '%',
+                (Math.min(difficulty - 2, 1) * 100).toFixed(0) + '%',
+                (Math.min(difficulty - 3, 1) * 100).toFixed(0) + '%',
+                (Math.min(difficulty - 4, 1) * 100).toFixed(0) + '%'
+            ])
+        } else {
+            setDifficultyEmote(undefined)
+            setDifficultyBars(['0%', '0%', '0%', '0%', '0%'])
+            setDifficultyHover('')
+        }
+    }, [difficulty])
+
+    useEffect(() => {
+        // Higher workload scores are more negative
+        if (workload) {
+            if (0 <= workload && workload < 3) {
```

Comment:
Maybe some of the difficulty and workload code could be combined into one function to avoid repetitive code.